packages/extension/src/contexts/LedgerContext/LedgerContext.tsx

```diff
@@ -729,44 +729,59 @@ const LedgerProvider: FC = ({ children }) => {
       signOnly = false
     ): Promise<SubmitTransactionResponse | SignTransactionResponse> => {
       const wallet = getCurrentWallet();
-      if (!client) {
-        throw new Error('You need to be connected to a ledger');
-      }
       if (!wallet) {
         throw new Error('You need to have a wallet connected');
       }
+      if (!signOnly) {
+        if (!client) {
+          throw new Error('You need to be connected to a ledger');
+        }
+      }
 
+      // Ensure the transaction has an Account or set it to the wallet's public address
       if (!payload.transaction.Account || payload.transaction.Account === '') {
         payload.transaction.Account = wallet.publicAddress;
       }
 
       // Validate the transaction
       validate(payload.transaction as unknown as Record<string, unknown>);
-      // Prepare the transaction
-      const prepared: Transaction = await client.autofill(payload.transaction);
+
+      // Prepare the transaction if needed
+      let prepared: Transaction = payload.transaction;
+      if (!signOnly && client) {
+        prepared = await client.autofill(payload.transaction);
+      }
+
       // Sign the transaction
       const signed = wallet.wallet.sign(prepared);
 
-      if (signOnly) {
-        if (!signed.tx_blob) {
-          throw new Error("Couldn't sign the transaction");
-        }
+      if (!signed.tx_blob) {
+        throw new Error("Couldn't sign the transaction");
+      }
 
+      // Handle signing-only case
+      if (signOnly) {
         return {
           signature: signed.tx_blob
         };
       }
 
-      // Submit the signed blob
+      // Ensure client is available for submission
+      if (!client) {
```

Comment:
already checked at the beginning

Reply:
Here we check it again in the case we are in the signOnly use case, this make sure that if we have another use case than the `signOnly` above, we still have the code working.
If I remove this line, I would have to write:
`const tx = await client?.submitAndWait(signed.tx_blob);`
I prefer reject it before and have a code stronger to resit to any feature possible bug.